File: backend/src/main/java/com/stayease/service/HotelService.java
package com.stayease.service;

import com.stayease.dto.request.HotelRequest;
import com.stayease.dto.response.HotelResponse;

import java.util.List;

public interface HotelService {

    HotelResponse addHotel(HotelRequest request);

    List<HotelResponse> getAllHotels();

    HotelResponse getHotelById(Long id);

    HotelResponse updateHotel(Long id, HotelRequest request);

    void deleteHotel(Long id);

    List<HotelResponse> searchByCity(String city);
}

File: frontend/src/services/hotelService.js
import api from "./api";

export const getAllHotels = () => {
    return api.get("/hotels");
};

export const getHotelById = (id) => {
    return api.get(`/hotels/${id}`);
};

export const addHotel = (hotel) => {
    return api.post("/hotels", hotel);
};

export const updateHotel = (id, hotel) => {
    return api.put(`/hotels/${id}`, hotel);
};

export const deleteHotel = (id) => {
    return api.delete(`/hotels/${id}`);
};

// ⭐ हे add कर
export const searchHotels = (city) => {
    return api.get(`/hotels/search?city=${city}`);
};